File: Examples/cosmicVoids/codes/sizeFunction.py
# ==========================================================================
# Example code: how to compute the theoretical size function of cosmic voids
# ==========================================================================

# to ensure compatibility in Python versions 2.x and 3.x
from __future__ import print_function

# import the CosmoBolognaLib #
import CosmoBolognaLib as cbl
import numpy as np

# set the CosmoBolognaLib and the current directories
cbl.SetDirs("../../../", "./")

# define a cosmological model, using default parameters #
cosm = cbl.Cosmology()

# Minimum and maximum of effective void radii
R_min = 1.
R_max = 30.

# number of radii at which the size function is computed
n_val = 10

# list of effective void radii with logarithmic binning
RR = np.logspace(np.log10(R_min), np.log10(R_max), n_val , endpoint=True)

# redshift of the sample
zz = 0.

# effective bias of the mass tracers
b_eff = 1.

print('The size function at z = %.2g'%zz ,' is :')

# compute the  Sheth & van de Weygaert size function 
for i in range(n_val):
    print('%.6e' %cosm.size_function(RR[i], zz, "SvdW", b_eff), '(h/Mpc)^3 at R = %.3g' %RR[i], ' Mpc/h ')

File: Examples/cosmicVoids/codes/sizeFunction.cpp
// ==========================================================================
// Example code: how to compute the theoretical size function of cosmic voids
// ==========================================================================

#include "Cosmology.h"
 
int main () {

  try {
     
    // -------------------------------------------------------------
    // ----- compute the Sheth & van de Weygaert size function -----
    // -------------------------------------------------------------
 
    // set the cosmology using Planck18 parameters 
    const cbl::cosmology::Cosmology Planck18 {cbl::cosmology::CosmologicalModel::_Planck18_};
    
    // logarithmic binning in the void effectiv radii
    const std::vector<double> radius = cbl::logarithmic_bin_vector(10, 1., 30.); 

    // redshift
    const double redshift  = 0.;

    // effective bias of the tracers (e.g. galaxies or galaxy clusters) used to detect the voids 
    const double b_eff = 1.; 


    // print the size function
    
    std::cout << "The size function at z = " << redshift << " is :" << std::endl;
    
    for (auto && rr : radius)
      std::cout << std::scientific << std::setprecision(6) << Planck18.size_function(rr, redshift, "SvdW", b_eff) << " (h/Mpc)^3 at R = " << std::defaultfloat << std::setprecision(3) << rr << " Mpc/h" << std::endl;

  }
  
  catch (cbl::glob::Exception &exc) { std::cerr << exc.what() << std::endl; exit(1); }
   
  return 0;
}
